### DISCUSSION TITLE
Feature request: [5.0] - add class to card body only for individual modules

### DISCUSSION THREAD

**Discussion by Scrabble96** [joomla/joomla-cms#43262] [Q&A]
### Is your feature request related to a problem? Please describe.

It's currently possible to:
1. Add a class to a module, including the header, i.e. the whole 'card' if using Cassiopeia
2. Add a class to a module header only, i.e. 'card-header'

but it is not possible to add a class to the module content/card-body only.

### Describe the solution you'd like
In my case I wanted to add (for one module only, not all modules of that type in that position), assign it to the 'side-bar' left position and then add a class (a version of grid-child or some other code) to the body of 'three-cols' so that the list on medium screens would display in four then three then two columns depending on screen width:

![mod-articles-category-with-four-cols](https://github.com/joomla/joomla-cms/assets/36987494/d28daa84-9222-41e8-84db-5f438e2f250e)

instead of one long single-column list with a large gap to the side:

![mod-articles-category-without-three-cols](https://github.com/joomla/joomla-cms/assets/36987494/8d6fdbbc-a43a-44ab-a7a2-17cc8665f4ec)

Or someone might want to apply a different background to that card body.

Yes, it could be done in user.css but would need to be done, I believe, for each individual module. Far easier to apply an existing class to whichever card requires it, _particularly if the user creating the module doesn't have access to the template files._

So, in the module setup under the Advanced tab, as well as 'Module Class' and 'Header Class' there would be a 'Body Class' and the 'card' chrome would reflect this with additional code after line 29 along the lines of:

```
$bodyClass            = htmlspecialchars($params->get('body_class', ''), ENT_QUOTES, 'UTF-8');
```
and line 57 would change to something like:
```
    <div class="card-body <?php echo $bodyClass; ?>">
```
[with a space after 'card-body' so that the class is rendered correctly]

I expect other additions would be needed.

**Comment by brianteeman** [comment]
Why not create your own chrome for this specific case?

**Reply by Scrabble96** [reply]
That would mean the user - not necessarily an admin- needing access to the template. IMHO this is not always a good idea.

**Reply by brianteeman** [reply]
not at all. what makes you think that. You have 3-5 chromes for a template - you probably only ever use 1 or 2. 
![image](https://github.com/joomla/joomla-cms/assets/1296369/b8a526cf-8ec4-4098-887e-d582040e24e2)

You as the admin create new chromes with the corporate colour scheme etc and then the person creating the module selects it from the list

![image](https://github.com/joomla/joomla-cms/assets/1296369/3286d16d-002d-4e83-a097-cfdae89ee3f8)

**Comment by Scrabble96** [comment]
Yes, I can see that I can select a custom chrome from the 'Module Style' [shouldn't this be called 'Module Chrome'?] but:

say there's custom CSS for a corporate colour to be used as the background in certain cases e.g.
```
.custom-colour {
  background-color: #d8b5ff;
}
```
For whatever reason, a user might want to apply that corporate background colour to the card body of a selected custom module, a newsflash module or any other type of module in different positions. Unless I understand incorrectly, your suggestion means having to create a custom chrome for each potential position and layout, instead of just applying 'custom-colour' to my proposed bodyClass field for the module content/card body.

**Comment by rdeutz** [comment]
Module chrome defines how a module gets rendered you can find the existing in layouts/chromes. Moving the issue to discussions, doesn't seens to me something we need to implement.
